test/test_visitor.py

```diff
@@ -0,0 +1,74 @@
+import itertools
+import typing
+from collections import defaultdict
+from collections.abc import Iterable
+
+import pytest
+from mypy.nodes import Node
+
+from refurb.visitor import METHOD_NODE_MAPPINGS, RefurbVisitor
+from refurb.visitor.mapping import VisitorNodeTypeMap
+from refurb.visitor.visitor import Checks
+
+
+@pytest.fixture
+def dummy_visitor() -> RefurbVisitor:
+    """
+    This fixture provides a RefurbVisitor instance with a visit method for each
+    possible node, but no checks to run.
+
+    This forces method generation but calling the methods does nothing.
+    """
+    checks: Checks = defaultdict(list)
+    checks |= {ty: [] for ty in METHOD_NODE_MAPPINGS.values()}
```

Comment:
I did try creating from the type alias, but I still had to write the default factory:

```python
checks = Checks(list)
```

That looked weird to me, it creates this question "What is list for" that makes you have to follow `Checks` to see what it is. This is why I went with the `defaultdict(list)` approach with the type annotation, I thought it would answer all questions on the spot.

What I didn't realise is that you could initialise the `defaultdict` with a dictionary as the second parameter.

So may I propose:
```python
    checks = defaultdict(
        list, {ty: [] for ty in METHOD_NODE_MAPPINGS.values()}
    )
    return RefurbVisitor(checks)
```

If you're not convinced, I'll go with your last suggestion, please let me know.

Reply:
I see you updated this to `checks: Checks = defaultdict(...)`, I presume because `checks = defaultdict(...)` doesn't include a type for `list`, am I correct there?

Either way, I think `checks = Checks(...)` would be more succinct, since you already need to type it as `Checks`, why not just initialize it with `Checks` as well?